File: src/test/java/org/goafabric/calleeservice/adapter/CalleeServiceAdapter.java
package org.goafabric.calleeservice.adapter;

import io.github.resilience4j.circuitbreaker.annotation.CircuitBreaker;
import org.goafabric.calleeservice.controller.dto.Callee;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.service.annotation.GetExchange;

@CircuitBreaker(name = "calleeservice")
public interface CalleeServiceAdapter {

    @GetExchange("/callees/sayMyName")
    Callee sayMyName(@RequestParam("name") String name);

    @GetExchange("/callees/sayMyOtherName/{name}")
    Callee sayMyOtherName(@PathVariable("name") String name);

}

File: src/test/kotlin/org/goafabric/calleeservice/adapter/CalleeServiceAdapter.kt
package org.goafabric.calleeservice.adapter

import org.goafabric.calleeservice.controller.dto.Callee
import org.springframework.beans.factory.annotation.Value
import org.springframework.stereotype.Component
import org.springframework.web.client.RestTemplate

@Component
class CalleeServiceAdapter(
    private val restTemplate: RestTemplate) {

    @Value("\${adapter.calleeservice.url}")
    private val url: String? = null

    fun sayMyName(name: String): Callee? {
        return restTemplate.getForObject("$url/callees/sayMyName?name={name}", Callee::class.java, name)
    }

    fun sayMyOtherName(name: String): Callee? {
        return restTemplate.getForObject("$url/callees/sayMyOtherName/{name}", Callee::class.java, name)
    }

}